action/protocol/staking/protocol.go

```diff
@@ -214,6 +217,9 @@ func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager
 	bcCtx := protocol.MustGetBlockchainCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
 	hu := config.NewHeightUpgrade(&bcCtx.Genesis)
+	if p.enableArchiveMode && p.hu.IsPost(config.Fairbank, blkCtx.BlockHeight) && p.hu.IsPre(config.Greenland, blkCtx.BlockHeight) {
+		p.saveStakingAddressHistory(blkCtx.BlockHeight, sm)
```

Comment:
also, i think should pass blkCtx.BlockHeight - 1?

for example BlockHeight = 5,500,001, this is before running this block, so at this time it reflects the balance at last block height 5,500,000

Reply:
and condition above should be
isPre(Greenland) && BlockHeight == GreenlandHeight(), so we can save the balance for GreenlandHeight - 1

Reply:
I'm not sure,but I'll change to height-1,and then will test with mainnet